File: vehicle-management/src/main/java/mk/ukim/finki/emt/vehiclemanagement/services/VehicleService.java
package mk.ukim.finki.emt.vehiclemanagement.services;

import mk.ukim.finki.emt.vehiclemanagement.domain.models.Vehicle;
import mk.ukim.finki.emt.vehiclemanagement.domain.models.VehicleId;
import mk.ukim.finki.emt.vehiclemanagement.domain.models.VehicleType;
import mk.ukim.finki.emt.vehiclemanagement.services.forms.VehicleForm;

import java.util.List;

public interface VehicleService {
    Vehicle findById(VehicleId id);
    Vehicle addVehicle(VehicleForm vehicleForm);
    List<Vehicle> getAll();
    Vehicle rent (VehicleId vehicleId);
    List<Vehicle> filter(String vehicleType, Double dailyPrice, String model);

}

File: react-car-rental/src/repository/VehicleService.js
import { vehicleService } from '../custom-axios/axios'

const VehicleService = {
    fetchVehicles: () => {
        return vehicleService.get('');
    },
    fetchVehicleById: (id) => {
        return vehicleService.get(`/${id}`);
    },
    addVehicle: (vehicle) => {
        return vehicleService.post('/add', vehicle);
    },
    filterVehicles: (type, model, dailyPrice) => {
        const params = {};
        if (type) params.type = type;
        if (model) params.model = model;
        if (dailyPrice) params.dailyPrice = dailyPrice;

        return vehicleService.get('/filter', { params });
    },
    fetchVehicleTypes: () => {
        return vehicleService.get('/types', {})
    }
}

export default VehicleService;
